internal/v2/command/project.go

```diff
@@ -54,6 +54,14 @@ func (r *CommandSide) getProjectByID(ctx context.Context, projectID, resourceOwn
 	return projectWriteModelToProject(projectWriteModel), nil
 }
 
+func (r *CommandSide) checkProjectExists(ctx context.Context, projectID, resourceOwner string) (bool, error) {
+	projectWriteModel, err := r.getProjectWriteModelByID(ctx, projectID, resourceOwner)
+	if err != nil {
+		return false, err
+	}
+	return projectWriteModel.State != domain.ProjectStateUnspecified && projectWriteModel.State != domain.ProjectStateRemoved, nil
+}
```

Comment:
i simplified checkUserExists like above so you don't have to handle a bool and error
if i matters if the error was thrown because of the state, you can check the error type